Bottom: 9effa49af131b4548fa521d5831cbf9e3870f3aa
Top:    51272879c99ba2f4ce836f2837091ed8ad2e1f15
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-05-17 09:53:41 +0100

Refresh of step-turn-off-inner-layers

---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index 87ed478..b18c2cd 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -1544,7 +1544,7 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
   pinvia_m_polyarea = info.poly;
 #endif
 
-  for (group = min_copper_group; group <= max_copper_group; group++)
+  for (group = min_copper_group; group <= max_copper_group; group += (max_copper_group - min_copper_group))
     {
       GList *new_objects;
       Coord depth = compute_depth (group);
@@ -1686,7 +1686,7 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
   // Repeat for each contour in the accumulated barrel M_POLYAERA
 
 
-  for (group = min_copper_group; group < max_copper_group; group++)
+  for (group = min_copper_group; group < max_copper_group; group += (max_copper_group - min_copper_group))
     {
       Coord top_depth;
       Coord bottom_depth;
@@ -1698,14 +1698,14 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
         break;
 
       /* Extrude barrel from group to group + 1 */
-      fprintf (stderr, "Extruding barrels from layer group %i to %i\n", group, group + 1);
+      fprintf (stderr, "Extruding barrels from layer group %i to %i\n", group, group + (max_copper_group - min_copper_group));
 
       g_assert (group_m_polyarea[group] != NULL);
-      g_assert (group_m_polyarea[group + 1] != NULL);
+      g_assert (group_m_polyarea[group + (max_copper_group - min_copper_group)] != NULL);
 
       /* Depth is the bottom? of the layer? */
       top_depth = compute_depth (group);
-      bottom_depth = compute_depth (group + 1);
+      bottom_depth = compute_depth (group + (max_copper_group - min_copper_group));
 
       barrel_objects = object3d_from_contours (barrel_m_polyarea,
 #ifdef REVERSED_PCB_CONTOURS
@@ -1746,7 +1746,7 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
           edge_ref e;
 
           POLYAREA *top_pa    = cntr_in_M_POLYAREA (pa->contours, group_m_polyarea[group    ] , false);
-          POLYAREA *bottom_pa = cntr_in_M_POLYAREA (pa->contours, group_m_polyarea[group + 1] , false);
+          POLYAREA *bottom_pa = cntr_in_M_POLYAREA (pa->contours, group_m_polyarea[group + (max_copper_group - min_copper_group)] , false);
 
           /* These conditions should not occur, and can likely only happen because some other
            * bug in the polygon processing code has created an inconsistent m_polyarea somewhere.
@@ -1976,7 +1976,7 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
   poly_Free (&barrel_m_polyarea);
   poly_Free (&drill_m_polyarea);
 
-  for (group = min_copper_group; group <= max_copper_group; group++)
+  for (group = min_copper_group; group <= max_copper_group; group += (max_copper_group - min_copper_group))
     {
       if (group_m_polyarea[group] != NULL)
         {
